lib/king_konf/variable.rb

```diff
@@ -15,7 +15,7 @@ def initialize(name:, type:, default: nil, description: "", required: false, all
 
     def cast(value)
       case @type
-      when :float then value.to_f
+      when :float then value.is_a?(String) || value.is_a?(Integer) ? value.to_f : value
```

Comment:
I'm not sure if @dasch wants this level of cleverness in the float parsing, but if so, you will need to add the same line to the `valid?` method. Note that the reason I added it to the symbol parsing was that symbols cannot be passed in from YAML. I don't think the same applies to floats.

Reply:
I think this makes sense – the problem was that we did `nil.to_f` on the default value, right?

Reply:
Exactly right @dasch